sudo_plugin/src/output/conv_facility.rs

```diff
@@ -0,0 +1,165 @@
+//! A module that implements the conversations API for a sudo plugin
+
+use crate::sys;
+
+use std::sync::{Arc, Mutex};
+use sudo_plugin_sys::{sudo_conv_t, sudo_conv_message, sudo_conv_reply};
+use std::io;
+use std::ffi::CString;
+use std::mem;
+use std::ptr;
+use std::slice;
+
+/// `ConvMsgType` is the type of conversation prompt as specified by 
+/// the sudo plugin
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum ConvMsgType {
+    /// Do not echo user input
+    PromptEchoOff  = sys::SUDO_CONV_PROMPT_ECHO_OFF,
+    /// Echo user input
+    PromptEchoOn   = sys::SUDO_CONV_PROMPT_ECHO_ON,
+    /// The prompt is an Error Message
+    ErrorMsg       = sys::SUDO_CONV_ERROR_MSG,
+    /// The prompt is an informational message
+    InfoMsg        = sys::SUDO_CONV_INFO_MSG,
+    /// Mask user input
+    PrompMask      = sys::SUDO_CONV_PROMPT_MASK,
+    /// Allows for echo if no TTY
+    PromptEchoOk   = sys::SUDO_CONV_PROMPT_ECHO_OK,
+    // This is only available on plugin version 1.14 which isn't supported yet
+    //ConvPreferTTY      = sys::SUDO_CONV_PREFER_TTY, /* flag: use tty if possible */
+}
```

Comment:
[I've pulled the `PrintFacility::Level` enum into the `output` module so it can be shared between these](https://github.com/square/sudo_pair/pull/72/files). Can you use that instead (`use crate::output::Level`)? The message doesn't actually end up making sense as an `enum` since it's really bitflags.

If/when we add support for turning echo on/off and masking prompts, I'll create a bitmask for those values.